### PR TITLE
Make @experimental final

### PR DESCRIPTION
**Pull request by noti0na1**
As discussed in #20223, this PR makes @experimental final.

Should unblock #20222.

### CODE REVIEW

**Review by nicolasstucki — COMMENTED** [review]
We should rebase https://github.com/scala/scala3/pull/20222 on this branch to ensure this change fixes the issue.

Otherwise LGTM

File: project/MiMaFilters.scala [hunk 1]
```diff
@@ -9,6 +9,7 @@ object MiMaFilters {
       // Additions that require a new minor version of the library
       Build.mimaPreviousDottyVersion -> Seq(
         ProblemFilters.exclude[DirectMissingMethodProblem]("scala.annotation.experimental.this"),
+        ProblemFilters.exclude[FinalClassProblem]("scala.annotation.experimental"),
```
Comment (nicolasstucki) [inline project/MiMaFilters.scala:12]: This will is a `BackwardsBreakingChanges`.

Comment (noti0na1) [reply project/MiMaFilters.scala]: Thanks!

**Comment by nicolasstucki** [comment]
@noti0na1 Could you squash and rebase?

File: project/MiMaFilters.scala [hunk 2]
```diff
@@ -56,6 +56,7 @@ object MiMaFilters {
           ProblemFilters.exclude[ReversedMissingMethodProblem]("scala.quoted.Quotes#reflectModule#MethodTypeModule.apply"),
           ProblemFilters.exclude[ReversedMissingMethodProblem]("scala.quoted.Quotes#reflectModule#MethodTypeMethods.methodTypeKind"),
           ProblemFilters.exclude[ReversedMissingMethodProblem]("scala.quoted.Quotes#reflectModule#MethodTypeMethods.isContextual"),
+          ProblemFilters.exclude[FinalClassProblem]("scala.annotation.experimental"),
```
Comment (nicolasstucki) [inline project/MiMaFilters.scala:59]: Maybe add a space or a comment as the comment above does not apply to this case.

**Comment by noti0na1** [comment]
@nicolasstucki #20222 has been rebased on to this PR and passes all the tests.

**Comment by noti0na1** [comment]
Closed, the change will be put together with #20222.

**Comment by joroKr21** [comment]
I extend `experimental` to cross-compile with Scala 2 (due to experimental APIs in `quotes.reflect`)
```scala
/** Used to cross-compile with Scala 2. */
class experimental extends scala.annotation.experimental
```

**Comment by sjrd** [comment]
Use a type alias instead.

**Pull request closed without merging**